### DISCUSSION TITLE
Sub ms precision for the Response Time

### DISCUSSION THREAD

**Discussion by facorazza** [artilleryio/artillery#1143] [Artillery]
Is it possible to get sub ms precision for the Response Time?

**Comment by hassy** [comment]
hi @facorazza 👋 There's no way to make Artillery's HTTP engine to report at sub-ms precision right now (though we do measure at nanosecond precision, it's the reporting that's done in milliseconds).

**Comment by MR-ERN** [comment]
Is there any update on this? Artillery seems really nice but I require sub ms precision.

**Reply by hassy** [reply]
No updates @MR-ERN. I'd be very curious to hear about your use-case that requires sub-ms precision, if there's any details you can share.

We do measure at nanosecond precision internally, so it should be fairly straightforward to add an option to Artillery to report in nanoseconds, but it's not something that many users have asked for, so we haven't prioritized it. If anyone wants to work on a PR for it though, we'd be happy to help!

**Reply by MR-ERN** [reply]
Working on my master's thesis at a tech company in finance looking at external tools for load testing for a time critical part of the system (low microseconds).

**Comment by rufreakde** [comment]
@hassy we are working in a big company with many microservices running a lot of extremly fast and small stateless services.

Many report values from 1 to 7 ms as results on performance tests. This means we do not have enough granularity to see if something changes on feature delivery. If it is rather simple to expose the more granular reports it would be much appreciated to be able to display management improvements even on that level.

Additionally it would be awesome to be able to exclude Ramp up and other "phases" from a run summary. Those outliers we dont want to have in our report.

**Reply by hassy** [reply]
@rufreakde thanks for the feedback! 

> Additionally it would be awesome to be able to exclude Ramp up and other "phases" from a run summary. Those outliers we dont want to have in our report.

Is the idea behind this to only include metrics from some phases in the final summary report? So that a warmup/ramp phase measurements don't affect the aggregate measurements (e.g. p99)?

**Reply by rufreakde** [reply]
Yes, if it would be possible set the „target“ phases to be reported in the aggregated summary. Declaratively that would be awesome so it does not need to be configured on the CLI command level but kn config level.
